virttest/utils_misc.py

```diff
@@ -2834,96 +2846,120 @@ def get_all_disks_did(session, partition=False):
+                 show in 'diskpart list disk'.
+    :param mountpoint: mount point list for all disks.
+    :param size: partition size.
+    :param fstype: filesystem type for the disk.
+    :param force: whether force format partition
+    :return a list: mount point list for all partitions.
+    """
+    cmd_header = 'echo list disk > disk &&'
+    cmd_header += 'echo select disk %s >> disk &&'
+    cmd_footer = '&& echo exit>> disk && diskpart /s disk'
+    cmd_footer += '&& del /f disk'
+    detail_cmd = 'echo detail disk >> disk'
+    detail_cmd = ' '.join([cmd_header, detail_cmd, cmd_footer])
+    set_rw_cmd = 'echo attributes disk clear readonly>> disk'
+    set_rw_cmd = ' '.join([cmd_header, set_rw_cmd, cmd_footer])
+    online_cmd = 'echo online disk>> disk'
+    online_cmd = ' '.join([cmd_header, online_cmd, cmd_footer])
+    if size:
+        size = int(float(normalize_data_size(size, order_magnitude="M")))
+        mkpart_cmd = 'echo create partition primary size=%s >> disk' % size
+    else:
+        mkpart_cmd = 'echo create partition primary >> disk'
+    mkpart_cmd = ' '.join([cmd_header, mkpart_cmd, cmd_footer])
+    format_cmd = 'echo list partition >> disk && '
+    format_cmd += 'echo select partition 1 >> disk && '
+    if not mountpoint:
+        format_cmd += 'echo assign >> disk && '
+    else:
+        format_cmd += 'echo assign letter=%s >> disk && '
+    format_cmd += 'echo format fs=%s quick >> disk '
+    format_cmd = ' '.join([cmd_header, format_cmd, cmd_footer])
+    mountpoint_list = []
+    for did in dids:
+        logging.info("Detail for 'Disk%s'" % did)
+        details = session.cmd_output(detail_cmd % did)
+        if re.search("Read.*Yes", details, re.I | re.M):
+            logging.info("Clear readonly bit on 'Disk%s'" % did)
+            session.cmd(set_rw_cmd % did)
+        pattern = "DISK %s.*Offline" % did
+        if re.search(pattern, details, re.I | re.M):
+            logging.info("Online 'Disk%s'" % did)
+            session.cmd(online_cmd % did)
+    for i, did in enumerate(dids):
+        details = session.cmd_output(detail_cmd % did)
+        if re.search(r"Volume.*%s" % fstype, details, re.I | re.M) and not force:
+            logging.info("Disk%s has been formated, cancel format" % did)
+            continue
+        logging.info("Create partition on 'Disk%s'" % did)
+        session.cmd(mkpart_cmd % did)
+        logging.info("Format the 'Disk%s' to %s" % (did, fstype))
+        if not mountpoint:
+            session.cmd(format_cmd % (did, fstype), timeout=300)
+            details = session.cmd_output(detail_cmd % did)
+            for line in details.splitlines():
+                pattern = "\s+Volume\s+\d+"
+                if re.search(pattern, line, re.I | re.M):
+                    mountpoint_list.append(line.split()[2])
+        else:
+            session.cmd(format_cmd % (did, mountpoint[i], fstype), timeout=300)
+    return mountpoint or mountpoint_list
```

Comment:
"return a list: mount point list for all partitions" in the doc, but
"return mountpoint or mountpoint_list" -- a boolean value in function.

Reply:
it is a list.
e.g.
>>> [ ] or [\"a\", \"b\"]
['a', 'b']
>>> [\"a\", \"b\"] or [ ]
['a', 'b']
>>>